test/tools/test_msginit.rb

```diff
@@ -224,6 +226,50 @@ def test_language_region_charset
       assert_equal(po_header(locale, language),
                    run_msginit("#{locale}.#{charset}"))
     end
+
+    def test_language_charset_with_replace_content_type
+      locale = "en"
+      assert_equal(po_header(locale, locale),
+                   run_msginit(locale, "CHARSET"))
+    end
+
+    def test_language_region_with_replace_content_type
+      locale = "en_US"
+      language = "en"
+      assert_equal(po_header(locale, language),
+                   run_msginit(locale, "CHARSET"))
+    end
+
+    def test_language_region_charset_with_replace_content_type
+      locale = "en_US"
+      language = "en"
+      charset = "UTF-8"
+      assert_equal(po_header(locale, language),
+                   run_msginit("#{locale}.#{charset}", "CHARSET"))
+    end
+  end
+
+  class TestCurrentCharset < self
+    def run_msginit(pot_charset)
+      create_pot_file("test.pot", charset: pot_charset)
+      po_file_path = "output.po"
+      @msginit.run("--output", po_file_path)
+      File.read(po_file_path)
+    end
+
+    def po_header(options)
+      super(current_locale, current_language, options)
+    end
+
+    def test_change
+      assert_equal(po_header(charset: "UTF-8"),
```

Comment:
```suggestion
      assert_equal(po_header(charset: Locale.current.charset),
```

We may want to define `current_charset` like other `current_*` methods.